### PR TITLE
Change cloudwatch submodule to pass context

### LINKED ISSUE #73: Cloudwatch submodule naming forces the var `function_name` to `lower`.
### Describe the Bug

I'm using this module to create a lambda, and I set the `name` attribute myself in CamelCase. The module creates all correctly _except_ the Cloudwatch log group, which is instead all lowercase. This of course doesn't match with the `function_name` setting, the function is unable to find its CW group, and the CW group (all lowercase) is basically there doing nothing.

This is likely due to the Cloudwatch submodule that, I couldn't figure out where TBH, is setting the `name` attribute to `lower`. The context is actually mentioning that setting defaults to `lower` but then the real `default` is `null`. If I use that module alone and I manually set `label_value_case` to `none` there, then I can get my names in upper/camel/pascal case.

My opinion is that is due to the `vars` being set to `module.this.vars` instead of the whole `context` being set to `module.this.context`. If I change that (into the tf cache, just for testing), my context is passed and `name` and all other attributes are consistent.

If you think this makes sense, I will create a PR to change that, it's literally a few characters to change :)

Thanks

### Expected Behavior

If the `function_name` variable is set to anything but `lowercase`, the Cloudwatch log group name should be consistent, it is instead forced to be all lowercase, that makes the function unable to use it.

### Steps to Reproduce

Just set `function_name` to `ThisIsMyLambda`, the module will create the Cloudwatch log group as `thisismylambda`.

### Screenshots

_No response_

### Environment

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by nnsense**
## what

A patch to pass the `context` instead of just `vars` to the CloudWatch submodule. 

## why

This module is only setting the context `vars` instead of the whole `context` inside the Cloudwatch submodule, leaving up to the submodule to set its own contex.

```
module "cloudwatch_log_group" {
  source  = "cloudposse/cloudwatch-logs/aws"
  version = "0.6.6"

  enabled = module.this.enabled

  iam_role_enabled  = false
  kms_key_arn       = var.cloudwatch_logs_kms_key_arn
  retention_in_days = var.cloudwatch_logs_retention_in_days
  name              = "/aws/lambda/${var.function_name}"
  tags              = module.this.tags
}
```

By settings the `context` (as advised by cloudposse) the root deployment are passed to the submodule, and an upper/camel/pascal case `function_name` will be consistent with the Cloudwatch group name.

## references

- Closes #73

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -15,7 +15,7 @@ module "cloudwatch_log_group" {
   kms_key_arn       = var.cloudwatch_logs_kms_key_arn
   retention_in_days = var.cloudwatch_logs_retention_in_days
   name              = "/aws/lambda/${var.function_name}"
-  tags              = module.this.tags
+  context           = module.this.context
```
Comment (milldr) [inline main.tf:18]: @nnsense since we're adding `context`, could you also remove `enabled`? It'll be included with `context`, so it would be redundant now

Comment (matteo-ssys) [reply main.tf:16]: Good point, done :)

Comment (nnsense) [reply main.tf:16]: Good point, done! :)

**Review by milldr — APPROVED** [review]

**Comment by nnsense** [comment]
Thanks @milldr :)

**Pull request merged**